content/docs/rfcs/12/README.md

```diff
@@ -61,77 +63,151 @@ The following are not considered as part of the adversarial model:
-`FilterRPC` this allows the node handler to determine how to handle a message as the Waku
-Filter protocol is not a request response based protocol but instead a push based system.
-
-The `requestId` MUST be a uniquely generated string. When a `MessagePush` is sent
-the `requestId` MUST match the `requestId` of the subscribing `FilterRequest` whose filters
-matched the message causing it to be pushed.
-
-#### FilterRequest
-
-A `FilterRequest` contains an optional topic, zero or more content filters and
-a boolean signifying whether to subscribe or unsubscribe to the given filters.
-True signifies 'subscribe' and false signifies 'unsubscribe'.
-
-A node that sends the RPC with a filter request and `subscribe` set to 'true' 
-requests that the filter node SHOULD notify the light requesting node of messages
-matching this filter.
-
-A node that sends the RPC with a filter request and `subscribe` set to 'false'
-requests that the filter node SHOULD stop notifying the light requesting node
-of messages matching this filter if it is currently doing so.
-
-The filter matches when content filter and, optionally, a topic is matched.
-Content filter is matched when a `WakuMessage` `contentTopic` field is the same.
-
-A filter node SHOULD honor this request, though it MAY choose not to do so. If
-it chooses not to do so it MAY tell the light why. The mechanism for doing this
-is currently not specified. For notifying the light node a filter node sends a
-MessagePush message.
-
-Since such a filter node is doing extra work for a light node, it MAY also
-account for usage and be selective in how much service it provides. This
-mechanism is currently planned but underspecified.
-
-#### MessagePush
-
-A filter node that has received a filter request SHOULD push all messages that
-match this filter to a light node. These [`WakuMessage`'s](./waku-message.md) are likely to come from the
-`relay` protocol and be kept at the Node, but there MAY be other sources or
-protocols where this comes from. This is up to the consumer of the protocol.
-
-A filter node MUST NOT send a push message for messages that have not been
-requested via a FilterRequest.
-
-If a specific light node isn't connected to a filter node for some specific
-period of time (e.g. a TTL), then the filter node MAY choose to not push these
-messages to the node. This period is up to the consumer of the protocol and node
-implementation, though a reasonable default is one minute.
+## Filter-Subscribe
+
+A filter service node MUST support the _filter-subscribe_ protocol
+to allow filter clients to subscribe, modify, refresh and unsubscribe a desired set of filter criteria.
+The combination of different filter criteria for a specific filter client node is termed a "subscription".
+A filter client is interested in receiving messages matching the filter criteria in its registered subscriptions.
+It is RECOMMENDED that filter service nodes allow only one subscription per client.
```

Comment:
Ah, I see the confusion that the current phrasing may lead to. B is indeed possible and the recommended way of doing it.

Note that:
> The combination of different filter criteria for a specific filter client node is termed a "subscription".

And this can be modified as you go along by subscribing and unsubscribing from content topics.

The idea is that a service node should not maintain multiple _sets_ of such filter criteria for the same client. This used to be the case with the previous spec, where the same client could register the same filter criteria several times at the same service node, each time using a different request ID. I think I should remove this confusing wording.

Reply:
Sorry my initial comment was not valid, I intended to delete it as I read the rest of the specs. I thin it's clear that Alice should update her subscription as I finished to read the spec.